languages/kotlin/generic/Parse_kotlin_tree_sitter.ml

```diff
@@ -1889,19 +1896,32 @@ and statements (env : env) ((v1, v2, v3) : CST.statements) =
   v1 :: v2
 
 and string_literal (env : env) (v1, v2, v3) : expr =
-  let v1 = token env v1 in
-  let v2 =
-    Common.map
-      (fun x ->
-        match x with
-        | `Str_content tok ->
-            (* string_content *)
-            Left3 (str env tok)
-        | `Interp x -> interpolation env x)
-      v2
-  in
-  let v3 = token env v3 in
-  G.interpolated (v1, v2, v3)
+  let l = token env v1 in
+  let r = token env v3 in
+  match v2 with
+  | [ `Interp (`DOLLAR_simple_id (v1, v2)) ] when in_pattern env ->
+      (* This is something of the form "$X". This is interpreted as an interpolated
+         string of a single identifier, but if this is the pattern,
+         it's probably not what the person writing the rule meant.
```

Comment:
Would appreciate something along the lines of "so instead we interpret this as ..."